src/main/java/in/ac/bits/protocolanalyzer/mvc/model/Experiment.java

```diff
@@ -78,4 +73,32 @@ private void init(String pcapPath) {
 		log.info("Successfully completed init method in session controller!!");
 	}
 
+	public void init(String pcapPath,String protocolGraphPath) throws Exception  
+	{
+	   checkFileAccess(protocolGraphPath);
+	   protocolGraphStr = new String(Files.readAllBytes(Paths.get(protocolGraphPath)));
+	   initWithPcapFileCheck(pcapPath, protocolGraphStr);  
+	}    
+	public void initWithPcapFileCheck(String pcapPath,String protocolGraphStr) throws Exception  
+	{
+	   checkFileAccess(pcapPath);
+	   this.pcapPath = pcapPath;
+	   this.protocolGraphStr = protocolGraphStr;
+       init(pcapPath);
+	}
+	
+	boolean checkFileAccess(String path) throws Exception{
```

Comment:
It is better to place checkFileAccess() method before initWithPcapFileCheck() method.

please write unit tests for these two methods.

Reply:
For unit tests, you can use mocks of all other dependencies like WebApplicationContext and Session.